File: ios/HelloFromTheOtherSide.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(HelloFromTheOtherSide, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(greet:(NSString*)name
                 resolver:(RCTPromiseResolveBlock)resolve
                 rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/HelloFromTheOtherSide.swift
@objc(HelloFromTheOtherSide)
class HelloFromTheOtherSide: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    }
    
    @objc(greet:resolver:rejecter:)
    func greet(name: String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) -> Void {
        resolve("Hello \(name)")
    }
}
